### ISSUE TITLE
libgen.h does not exist on Windows

### ISSUE THREAD

**Issue by willpatera** [pupil-labs/pyndsi#23]
`libgen.h` does not exist on Windows. Either we do a C++ re-implementation of what we need or find a workaround.

**Comment by willpatera** [comment]
@saki4510t - can you provide a solution/workaround so that pyndsi can build on windows?

**Comment by saki4510t** [comment]
@willpatera I will try.

**Comment by saki4510t** [comment]
Hi @willpatera
I can build pyndsi (current master branch) on Windows10 with vc2015 and create wheel package now although there are still lot of warnings.
Unfortunately I still can't run `run_capture.bat` and I'm not sure it works or not. Is there any other way to test ndsi module?

**Comment by saki4510t** [comment]
Hi @willpatera
I add modification to pyndsi and can compile on my Windows 10. Unfortunately I still can't build pupil capture on Windows and can't test it. So I just make patch file instead of pull request.
If you have time, please try and test it.

[patch.zip](https://github.com/pupil-labs/pyndsi/files/931620/patch.zip)

**Comment by willpatera** [comment]
@saki4510t - I will modify the Windows build instructions - I'm currently re-installing all dependencies. 

I will try your patch tomorrow with the current build env. Thanks 😄

**Comment by willpatera** [comment]
Hi @saki4510t I was able to apply the patch and able to build pyndsi after a few changes to setup.py with your patch. 

Please make a PR for your changes. 

I made minor changes to setup.py:

```python
if platform.system() == 'Darwin':
    include_dirs += ['/usr/local/opt/jpeg-turbo/include/']
    libs += ['turbojpeg']
    library_dirs += ['/usr/local/opt/jpeg-turbo/lib/']
    libs += ['avutil', 'avformat', 'avcodec', 'swscale']
elif platform.system() == 'Linux':
    libs = ['rt', 'turbojpeg']
    libs += ['avutil', 'avformat', 'avcodec', 'swscale']
elif platform.system() == 'Windows':
    libs = ['winmm']
    tj_dir = 'C:\\work\\libjpeg-turbo-VC64'
    tj_lib = tj_dir + '\\lib\\turbojpeg.lib'
    include_dirs += [tj_dir + '\\include']
    extra_objects += [tj_lib]
    include_dirs += ['ndsi\\h264\\windows']
    ffmpeg_libs = 'C:\\work\\ffmpeg-3.2-win64-dev\\lib'
    include_dirs += ['C:\\work\\ffmpeg-3.2-win64-dev\\include']
    libs += [ffmpeg_libs+'\\avutil',ffmpeg_libs+'\\avformat',ffmpeg_libs+'\\avcodec',ffmpeg_libs+'\\swscale']
```

**Comment by saki4510t** [comment]
@willpatera 
OK, I will.

BTW, Is it possible to remove absolute path `C:\\work` and use relative path, ex. by adding dependancies like libjpeg-turbo-VC64, ffmpeg and ceres etc. under `{project root}/pupil_external`?
Because there are no description about `c:work` in `Pupil Docs` and removing absolute path can avoid manual modification of `setup.py`.

Regarding a constructing build environment, did you try to use `msys2`?

**Comment by willpatera** [comment]
closed with #24
